lte/gateway/c/session_manager/SessionState.h

```diff
@@ -159,20 +160,31 @@ class SessionState {
   /* methods of new messages of 5G and handle other message*/
   uint32_t get_current_version();
 
+  /* method to set update the session current version */
   void set_current_version(
       int new_session_version, SessionStateUpdateCriteria& uc);
 
-  void insert_pdr(SetGroupPDR* rule);
+  /* method to add new PDR rules to session */
+  void insert_pdr(
+      SetGroupPDR* rule, SessionStateUpdateCriteria& session_uc, bool crit_add);
 
-  void set_remove_all_pdrs();
+  /* method to change the PDR state */
+  void set_all_pdrs(enum PdrState);
+  int32_t pdr_index(uint32_t pdr_index);
+  /* method to search specific pdr id existense */
+  bool search_pdr(unsigned int id);
```

Comment:
pdr_index already delivers the needed search functionality. you can eliminate this...

Reply:
It used in different scenariaos, so kept .